File: NetworkAccess/library/src/main/java/jv/lee/library/core/NetworkCallbackImpl.java
package jv.lee.library.core;

import android.net.ConnectivityManager;
import android.net.Network;
import android.net.NetworkCapabilities;
import android.util.Log;

import jv.lee.library.utils.Constants;

/**
 * @author jv.lee
 * @date 2019/9/23.
 * @description
 */
public class NetworkCallbackImpl extends ConnectivityManager.NetworkCallback {
    @Override
    public void onAvailable(Network network) {
        super.onAvailable(network);
        Log.d(Constants.LOG_TAG, "网络连接:" + network.toString());
    }

    @Override
    public void onLost(Network network) {
        super.onLost(network);
        Log.d(Constants.LOG_TAG, "网络丢失:" + network.toString());
    }

    @Override
    public void onCapabilitiesChanged(Network network, NetworkCapabilities networkCapabilities) {
        super.onCapabilitiesChanged(network, networkCapabilities);
        if (networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_VALIDATED)) {
            //网络发生了变更
            if (networkCapabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI)) {
                Log.i(Constants.LOG_TAG, "网络发生了变更，当前网络类型是wifi");
            } else {
                Log.i(Constants.LOG_TAG, "网络发生了变更，当前网络类型是");
            }
        }
    }

    @Override
    public void onUnavailable() {
        super.onUnavailable();
    }
}

File: KotlinLibraryX/app/src/main/java/com/lee/app/core/NetworkCallbackImpl.kt
package com.lee.app.core

import android.net.ConnectivityManager
import android.net.Network
import android.net.NetworkCapabilities
import com.lee.library.utils.LogUtil

/**
 *
 * @author jv.lee
 * @date 2020-03-07
 *
 */
class NetworkCallbackImpl : ConnectivityManager.NetworkCallback() {

    var firstCode = 0
    var lastCode = 0

    override fun onAvailable(network: Network) {
        super.onAvailable(network)
        firstCode = network.toString().toInt()
        LogUtil.i("${network}网络：连接成功")
    }

    override fun onLost(network: Network) {
        super.onLost(network)
        lastCode = network.toString().toInt()
        if (lastCode >= firstCode) {
            LogUtil.i("${network}网络：断开连接")
        }
    }

    override fun onCapabilitiesChanged(
        network: Network,
        networkCapabilities: NetworkCapabilities
    ) {
        super.onCapabilitiesChanged(network, networkCapabilities)
        if (networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_VALIDATED)) {
            if (networkCapabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI)) {
                LogUtil.d("onCapabilitiesChanged: 网络类型为wifi :$network")
            } else if (networkCapabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)) {
                LogUtil.d("onCapabilitiesChanged: 蜂窝网络:$network")
            } else {
                LogUtil.d("onCapabilitiesChanged: 其他网络:$network")
            }
        }
    }

}